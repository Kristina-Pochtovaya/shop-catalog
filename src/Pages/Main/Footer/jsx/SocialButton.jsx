import React from 'react';

export const SocialButton = ({type}) => (
    <a href="" className={type === "facebook" ? "fa fa-facebook" : "fa fa-instagram"}   >

    </a>

);



