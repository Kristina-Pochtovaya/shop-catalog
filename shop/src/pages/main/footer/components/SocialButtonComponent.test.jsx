import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import SocialButton from './SocialButtonComponent';

configure({ adapter: new Adapter() });

test('should render  Social Button Component with className=facebook, if prop(type)=facebook', () => {
  const socialButton = shallow(<SocialButton type="facebook" />);

  expect(toJson(socialButton)).toMatchSnapshot();
  expect(socialButton.find('.fa').hasClass('fa-facebook')).toEqual(true);
  expect(socialButton.find('.fa').hasClass('fa-instagram')).toEqual(false);
});

test('should render  Social Button Component with className=instagram, if prop(type)=instagram', () => {
  const socialButton = shallow(<SocialButton type="instagram" />);

  expect(socialButton.find('.fa').hasClass('fa-facebook')).toEqual(false);
  expect(socialButton.find('.fa').hasClass('fa-twitter')).toEqual(false);
  expect(socialButton.find('.fa').hasClass('fa-instagram')).toEqual(true);
});
