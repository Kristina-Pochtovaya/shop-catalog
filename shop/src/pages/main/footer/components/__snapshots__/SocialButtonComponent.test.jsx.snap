// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render  Social Button Component 1`] = `
<link
  className="fa fa-instagram"
  href=""
/>
`;

exports[`should render  Social Button Component with className=facebok, if prop=facebook 1`] = `
<link
  className="fa fa-facebook"
  href=""
/>
`;

exports[`should render  Social Button Component with className=facebook, if prop(type)=facebook 1`] = `
<link
  className="fa fa-facebook"
  href=""
/>
`;

exports[`should render  Social Button Component with className=facebook, if prop=facebook 1`] = `
<link
  className="fa fa-facebook"
  href=""
/>
`;

exports[`should render an Header with button Назад 1`] = `
<link
  className="fa fa-instagram"
  href=""
/>
`;
